### PR TITLE
feat: Deprecate `datasets` module, rename to `inferences`

### LINKED ISSUE #2732: [ENHANCEMENT] rename Dataset to Inferences
Need to carve out room for future datasets. We should deprecate the old datasets and rename them

### PR DESCRIPTION
**Pull request by anticorrelator**
Resolves #2732 

- Deprecates `phoenix.datasets` module, interfaces are available under `phoenix.inferences`
- `px.Dataset` class renamed to `px.Inference`
- `px.ExampleDatasets` class renmaed to `px.ExampleInferences`
- `Datasets.from_open_inference` deprecated and removed from `Inference`

### CODE REVIEW

**Review by RogerHYang — APPROVED** [review]

File: tests/server/api/types/test_dataset.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 import pytest
 from pandas import DataFrame, Timestamp
-from phoenix.datasets.dataset import Dataset as InternalDataset
-from phoenix.datasets.dataset import Schema
+from phoenix.inferences.inference import Inference as InternalDataset
```
Comment (axiomofjoy) [inline tests/server/api/types/test_dataset.py:5]: Rename to `InternalInference`.

File: src/phoenix/core/model.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-from phoenix.datasets.dataset import Dataset
-from phoenix.datasets.schema import EmbeddingColumnNames, EmbeddingFeatures
+from phoenix.inferences.inference import Inference
+from phoenix.inferences.schema import EmbeddingColumnNames, EmbeddingFeatures
 
 from .embedding_dimension import EmbeddingDimension
 
 
 def _get_embedding_dimensions(
```
Comment (axiomofjoy) [inline src/phoenix/core/model.py:9]: I think we can rename these arguments for clarity.

File: src/phoenix/core/model_schema_adapter.py [hunk 3]
```diff
@@ -132,8 +132,8 @@ def create_model_from_datasets(*datasets: Optional[Dataset]) -> Model:
     )
 
 
-def _is_dataset(obj: Optional[Dataset]) -> TypeGuard[Dataset]:
-    return type(obj) is Dataset
+def _is_dataset(obj: Optional[Inference]) -> TypeGuard[Inference]:
+    return type(obj) is Inference
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema_adapter.py:136]: These probably deserve to be renamed.

File: src/phoenix/server/main.py [hunk 4]
```diff
@@ -114,9 +114,9 @@ def _load_items(
     trace_dataset_name: Optional[str] = None
     simulate_streaming: Optional[bool] = None
 
-    primary_dataset: Dataset = EMPTY_DATASET
```
Comment (axiomofjoy) [inline src/phoenix/server/main.py:117]: Probably can be renamed.

File: tests/datasets/test_inference.py [hunk 5]
```diff
@@ -12,17 +12,17 @@
...
-    Dataset,
+from phoenix.inferences.errors import DatasetError
+from phoenix.inferences.inference import (
+    Inference,
     _normalize_timestamps,
     _parse_dataframe_and_schema,
 )
-from phoenix.datasets.errors import DatasetError
-from phoenix.datasets.schema import (
+from phoenix.inferences.schema import (
```
Comment (axiomofjoy) [inline tests/datasets/test_inference.py:25]: Looks like the test names in this file are still named with `dataset` names.

File: tests/server/api/types/test_inference.py [hunk 6]
```diff
@@ -28,14 +28,14 @@ def input_dataset() -> InternalDataset:
...
+    return InternalInference(dataframe=input_df, schema=input_schema)
 
 
 class TestDatasetEvents:
     def test_no_input_dimensions_correctly_selects_event_ids_and_all_features_and_tags(
         self,
-        primary_dataset: InternalDataset,
-        reference_dataset: InternalDataset,
+        primary_dataset: InternalInference,
+        reference_dataset: InternalInference,
```
Comment (axiomofjoy) [inline tests/server/api/types/test_inference.py:38]: looks like this file still has a lot of dataset names

File: src/phoenix/inferences/fixtures.py [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+    wide_data_fixture,
+    deep_data_fixture,
+    llm_summarization_fixture,
+    wikipedia_fixture,
+    chatbot_fixture,
+)
+NAME_TO_FIXTURE = {fixture.name: fixture for fixture in FIXTURES}
+
+
+def get_datasets(
```
Comment (axiomofjoy) [inline src/phoenix/inferences/fixtures.py:416]: This can be renamed.

File: src/phoenix/inferences/fixtures.py [hunk 8]
```diff
@@ -0,0 +1,560 @@
...
+    fixture_name: str,
+    no_internet: bool = False,
+) -> Tuple[Inference, Optional[Inference], Optional[Inference]]:
+    """
+    Downloads primary and reference datasets for a fixture if they are not found
```
Comment (axiomofjoy) [inline src/phoenix/inferences/fixtures.py:421]: same here

File: src/phoenix/inferences/inference.py [hunk 9]
```diff
@@ -0,0 +1,523 @@
...
+)
+from .validation import validate_dataset_inputs
+
+logger = logging.getLogger(__name__)
+
+# A schema like object. Not recommended to use this directly
+SchemaLike: TypeAlias = Any
+
+
+class Inference:
```
Comment (axiomofjoy) [inline src/phoenix/inferences/inference.py:37]: Should this be plural because it represents multiple inferences?

**Review by axiomofjoy — APPROVED** [review]
`Inference` vs. `Inferences`?

Just checking the diff, I'm still seeing a large number of old `dataset` names.

**Review by mikeldking — CHANGES_REQUESTED** [review]
I think it should be plural. Also can we cascade the notebook changes? We need to wait until the release. You will also have to pin the notebooks lowerbound

**Review by mikeldking — APPROVED** [review]
Approving to unblock

**Pull request merged**